contrib/dracut/90zfs/zfs-env-bootfs.service.in

```diff
@@ -11,12 +11,7 @@ ExecStart=/bin/sh -c '
     . /lib/dracut-zfs-lib.sh;                                                                  \
     decode_root_args || exit 0;                                                                \
     [ "$root" = "zfs:AUTO" ] && root="$(@sbindir@/zpool list -H -o bootfs | grep -m1 -vFx -)"; \
-    rootflags="$(getarg rootflags=)";                                                          \
-    case ",$rootflags," in                                                                     \
```

Comment:
lifting this to a single-use function with a weird calling convention doesn't really make sense here, just do
```sh
[ "$(@sbindir@/zfs get -H -o value mountpoint "$root")" = legacy ] || case ",$rootflags," in
```
(or re-spell with an if instead of ||)